### ISSUE TITLE
App crash of mpv.exe with module lua51.dll

### ISSUE THREAD

**Issue by francwalter** [57op/simple-mpv-webui-windows-libs#4]
Hallo
since some weeks I cannot start mpv.exe (actual version or older) anymore with simple-mpv-webui script on my Windows 10 Pro 21H2 (64-Bit, 19044.1682) PC.
If I start mpv, nothing happens, if I start it on commandline with a movie as path I dont see any error, even not with -v parameter.
After some seconds, I am back on the command prompt, no mpv started.
I get an app crash of mpv.exe which I only see in Windows Event Viewer:

```
Name der fehlerhaften Anwendung: mpv.exe, Version: 2.0.0.0, Zeitstempel: 0x00000000
Name des fehlerhaften Moduls: lua51.dll, Version: 0.0.0.0, Zeitstempel: 0x00000000
Ausnahmecode: 0xc0000005
Fehleroffset: 0x00000000000462a2
ID des fehlerhaften Prozesses: 0x2038
Startzeit der fehlerhaften Anwendung: 0x01d85f96d25f6cd3
Pfad der fehlerhaften Anwendung: C:\Program Files\mpv\mpv.exe
Pfad des fehlerhaften Moduls: C:\Program Files\mpv\lua51.dll
Berichtskennung: 74130372-d2ad-4424-ac91-c9001d3049d2
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist: 
```

If I rename lua51.dll or the lua folder (in mpv folder) then I get an error about missing lua but mpv starts, doesnt crashes.

I dont know what changed or how I could fix this. The lua51.dll is unchanged, maybe some Windows Update did this?

Another Windows 10 PC with the same simple-mpv-webui script and this lua package installed, works without errors.

Thank, frank

**Comment by francwalter** [comment]
@57op no clue what this error and crash could be?

**Comment by SonaliBendre** [comment]
doesn;t work in windows

**Comment by 57op** [comment]
sorry for the delay, I'm working on this and will update on any progress

**Comment by francwalter** [comment]
Would be very nice to fix, if it is a bug in your dll, thanks!

**Comment by 57op** [comment]
i updated upstream sources of luajit and luasocket in the hope of fixing this issue, but unfortunately it didn't solve anything.

on their own, luajit and luasocket work fine.
i assume something must've changed within mpv (of course), sadly i don't have the proper amount of time to investigate this thoroughly right now

**Comment by francwalter** [comment]
You mean with an older mpv version it could still work?
I will try it...
I don't need actual mpv but really love the feature with webui on windows, very useful.
Will test an older release of mpv and comment again here...
Thank already much appreciated that help!

**Comment by 57op** [comment]
>You mean with an older mpv version it could still work?

yes, i've been using [20211003 build x64](https://sourceforge.net/projects/mpv-player-windows/files/64bit/mpv-x86_64-20211003-git-5304e9f.7z/download) for many months and the extension works fine

**Comment by francwalter** [comment]
Oh, great!
Indeed, that works again!
That is good, very good :)
Thank a lot, this helps!
This is Properties with the working mpv version:

![grafik](https://user-images.githubusercontent.com/927269/171996175-301a178a-0b0c-4437-8a12-5a4bb52efad1.png)

Version 0.33.0-349

**Comment by francwalter** [comment]
By the way: this is the debug when crash with actual mpv.exe in Windows (with Visual Studio 2012 Debugger):

![grafik](https://user-images.githubusercontent.com/927269/172000769-c6557d13-0dd4-4341-8da8-0e528c1d1533.png)

But I dont think that it is of any help 😏

**Comment by 57op** [comment]
hey, i've published a [new release](https://github.com/57op/simple-mpv-webui-windows-libs/releases/tag/0.0.2).
it's working for me on the newest mpv build ([20220529](https://sourceforge.net/projects/mpv-player-windows/files/64bit/mpv-x86_64-20220529-git-afe2902.7z/download)) by shinchiro.

if you can test and report here, i would be grateful

**Comment by francwalter** [comment]
It doesnt work with the 0.34.0-302-gafe29026ed from 29.05.2022 (which comes from the updater.bat).
But this 002 doesnt work anymore with the old mpv from 03.10.2021 which works with 001.
Sorry 😢

**Comment by 57op** [comment]
thanks for the report 😄 
sorry, you are not supposed to use the newer DLLs on older mpv builds 😆 

anyway, i noticed i was using an older release of simple-mpv-webui (namely v2.1.0).
i switched to the latest one v3.0.0 and it doesn't work for me as well.

it seems to me there's some issues with ipv6 address binding.
so if it has no use to you, you can temporary fix this by disabling ipv6 using webui script options:
`script-opts=webui-ipv6=no`

basically, add the above snippet to your mpv.conf

anyway, this issue will stay open until ipv6 is fixed as well

edit:
it seems really unstable on v3.0.0, even when using ipv4 only..

**Comment by 57op** [comment]
i updated the v0.0.2 release, now it should work even with ipv6

**Comment by open-dynaMIX** [comment]
@57op Thanks a lot for your work on this!

BTW: In the webui [README](https://github.com/open-dynaMIX/simple-mpv-webui#windows) this repo is linked, with `chuck-` as username, but it redirects here. Is this due to a username change? If so, I'd like to update those links.

**Comment by francwalter** [comment]
> i updated the v0.0.2 release, now it should work even with ipv6

Thank a very lot! Now it works with the actual version of mpv.exe (and actual simple-mpv-webui).
I close the issue then, thank!

**Comment by 57op** [comment]
you're welcome, happy to know it's working for you too!

> BTW: In the webui [README](https://github.com/open-dynaMIX/simple-mpv-webui#windows) this repo is linked, with `chuck-` as username, but it redirects here. Is this due to a username change? If so, I'd like to update those links.

please do, i switched to a different username some time ago due to some issues with `github.io` subdomain.
now that you mentioned it, i should update my README as well.

thanks for your work on simple-mpv-webui :)
